src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3399,7 +3400,14 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
 	bool supported = true;
 	bool allow_privileged_sock_opt = false;
 
+	if (m_sock_state <= TCP_SOCK_ACCEPT_READY && __optval != NULL && is_inherited_option(__level, __optname))
+		m_socket_options_list.push_back(new socket_option_t(__level, __optname,__optval, __optlen));
+
 	if ((ret = sockinfo::setsockopt(__level, __optname, __optval, __optlen)) != SOCKOPT_PASS_TO_OS) {
+		if (ret == SOCKOPT_INTERNAL_VMA_SUPPORT &&
+		    m_sock_state <= TCP_SOCK_ACCEPT_READY && __optval != NULL &&
+		    is_inherited_option(__level, __optname))
+			m_socket_options_list.push_back(new socket_option_t(__level, __optname,__optval, __optlen));
```

Comment:
two push_back() calls ?

Reply:
The logic is not good.
It will break non-vma parameters such as SO_TIMESTAMPING/IP_TTL/SO_TIMESTAMP/SO_TIMESTAMPNS.

Reply:
and all the rest of the TCP socket options from sockinfo_tcp...